### PR TITLE
Release 0.5.1-beta

### PR DESCRIPTION
**Pull request by tsterbak**
This release adds:
-  Allow flashing of the vendor_boot partition
-  Add --disable-verity --disable-verification to flash vbmeta command

New devices:
- support for begonia (Xiaomi Redmi Note 8 Pro) (thanks to @anon1892)
- support for on7xelte (Samsung Galaxy J7 Prime) (thanks to @MagicLike)
- support for a7y17lte (Samsung Galaxy A7 2017) (thanks to @SirRGB)
- support for jfltexx (Samsung Galaxy S4) (thanks to @MagicLike)
- support for dipper (Xiaomi Mi 8) (thanks to @anon1892)
- support for surya/karna (Poco X3 - X3 NFC)  (thanks to @anon1892)
- support for alioth (Redmi K40 / Mi 11X / Poco F3) (thanks to @anon1892)
- new device code variant to `j7elte` & added TWRP link (thanks to @MagicLike)

Other improvements:
- updates for some OnePlus and Pixel configs with make use of the flashing of additional partitions if needed for Android 13
- a requirement checkbox to encourage booting stock OS at least once
- migrate xda developers link to a new url
- misc docs and minor config corrections/additions (thanks to @SirRGB)

### CODE REVIEW

**Comment by anon1892** [comment]
Just a little fix : `vayu` have been successfully tested (#270)
So the config have to be updated : https://github.com/openandroidinstaller-dev/openandroidinstaller/blob/95e08f7328e67347e1d423acba921a1ffad4bd2f/openandroidinstaller/assets/configs/vayu.yaml#L10 (it's already done for README)

File: openandroidinstaller/assets/configs/dipper.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  device_name: Xiaomi Mi 8
+  is_ab_device: false
+  device_code: dipper
+  supported_device_codes:
+    - dipper
+  notes:
+    - You should install Android 10 or newer ROM.
+requirements:
+  android: 10 (MiUI 12)
+untested: true
```
Comment (anon1892) [inline openandroidinstaller/assets/configs/dipper.yaml:13]: Oops, wrong alignment for this metadata

File: docs/building_the_application_for_your_platform.md [hunk 2]
```diff
@@ -6,9 +6,8 @@ If you build the application for your platform and want to contribute the build,
 
 ## Run OpenAndroidInstaller for development
 
-Currently development is only supported on Ubuntu Linux. MacOS and Windows should also work fine. You might need to install additional USB-drivers on Windows.
+Currently development is only supported on Ubuntu Linux. MacOS, Windows as well as other Linux distributions supporting flatpak should also work fine. You might need to install additional USB-drivers on Windows.
```
Comment (anon1892) [inline docs/building_the_application_for_your_platform.md:9]: Works fine in Fedora too

**Comment by MagicLike** [comment]
https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/317/files/53d836b4828f022fd0d89371ac3b0265c3201bc0#r1377870986

**Comment by tsterbak** [comment]
> 

> fastboot reboot recovery wasn't changed to fastboot reboot-recovery here. Is this an oversight?

Should it be changed in general? I can change it in the function if you want :) I wasn't sure about it.

> Else, LGTM

Great! Thanks for the quick review :)

**Comment by tsterbak** [comment]
> Should it be changed in general? I can change it in the function if you want :) I wasn't sure about it.

Both commands seem to be supported by fastboot, but only `reboot recovery` is in the specification as far as I can tell. But they should be largely equivalent.

**Comment by tsterbak** [comment]
> fastboot reboot recovery wasn't changed to fastboot reboot-recovery here. Is this an oversight?

Should be fine, I changed it now :)

**Comment by tsterbak** [comment]
I will give this some more testing tomorrow and then release it :)

Thanks for the great work everyone!

File: README.md [hunk 3]
```diff
@@ -181,11 +184,15 @@ Vendor | Device Name | CodeName | Models | Status
 Xiaomi | Redmi Note 7 | [lavender](https://wiki.lineageos.org/devices/lavender) |  | tested
 Xiaomi | Redmi 7A / 8 / 8A / 8A Dual | [Mi439](https://wiki.lineageos.org/devices/Mi439) : pine / olive / olivelite / olivewood | | tested
 Xiaomi | Redmi Note 8 / 8T | [ginkgo](https://wiki.lineageos.org/devices/ginkgo) / willow |  | untested
+Xiaomi | Redmi Note 8 Pro | begonia |  | untested
+Xiaomi | Mi 8 | [dipper](https://wiki.lineageos.org/devices/dipper) |  | untested
 Xiaomi | Redmi 9A / 9C / 9AT / 9i / 9A Sport / 10A / 10A Sport | garden / dandelion / blossom / angelican | | tested
-Xiaomi | Mi 9T / Redmi K20 | [davinci](https://wiki.lineageos.org/devices/davinci) / davinciin |  | untested
+Xiaomi | Mi 9T / Redmi K20 | [davinci](https://wiki.lineageos.org/devices/davinci) / davincin |  | untested
```
Comment (anon1892) [inline README.md:190]: Still not good :wink:

**Comment by anon1892** [comment]
> Add --disable-verity --disable-verification to flash vbmeta command

What does that change? It's not needed for actually supported Xiaomi devices, afaik

File: README.md (hunk 3 above)
Comment (tsterbak) [reply README.md]: But now, I hope :grimacing:

**Comment by anon1892** [comment]
Ok, thanks!

**Comment by MagicLike** [comment]
I'd really like to have #337 included...

**Comment by tsterbak** [comment]
> I'd really like to have https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/337 included...

Yeah, the changes would be nice. But it's not blocking for the release, since changes to the README can me done without a release.

**Review by MagicLike — APPROVED** [review]
LGTM!

**Pull request merged**